tests/rest/admin/test_user.py

```diff
@@ -338,7 +338,7 @@ def test_register_mau_limit_reached(self):
 
         # Set monthly active users to the limit
         store.get_monthly_active_count = Mock(
-            return_value=defer.succeed(self.hs.config.max_mau_value)
+            side_effect=lambda: defer.succeed(self.hs.config.max_mau_value)
```

Comment:
The difference is that it creates a new value for each call (instead of re-using the same `Deferred`). I don't remember why this was necessary though. I'll check again!

Reply:
ah no worries. It sounds like a sensible change.

Reply:
This change wasn't necessary. Not sure why I did it. I *think* I had originally changed these from `defer.succeed` to `make_awaitable`, which requires a change like this since an awaitable isn't re-usable.

I'll revert it to keep it simple for now.